File: dotnet/Yitisna.Study.VBStudy/Yitian.Study.VBStudy.Basic/ExceptionHandling.vb
Public Module ExceptionHandling
    Sub HandleException()
        Try
            Throw New ArithmeticException("除数不能为零")
        Catch ex As ArithmeticException
            Console.WriteLine($"捕获了数学计算异常:{ex.Message}")
        Catch ex As Exception
            Console.WriteLine($"捕获了异常:{ex.Message}")
        Finally

            Console.WriteLine("终结代码")
        End Try
    End Sub
End Module

File: java/java-samples/src/main/java/tech/techstay/exception/ExceptionHandling.java
package tech.techstay.exception;

public class ExceptionHandling {
  public static void main(String[] args) {
    try {
      throw new MyException("throwing some custom exception");
    } catch (ArithmeticException | NumberFormatException e) {
      System.out.println("arithmetic exception caught");
    } catch (MyException e) {
      e.printStackTrace();
      System.out.println("custom exception caught!");
    } catch (Exception e) {
      System.out.println("catching the rest exceptions");
      e.printStackTrace();
    } finally {
      System.out.println("doing some cleaning tasks here");
    }
  }
}


class MyException extends RuntimeException {
  MyException(String msg) {
    super(msg);
  }
}
